phalcon/Mvc/Model.zep

```diff
@@ -888,25 +885,17 @@ abstract class Model implements EntityInterface, ModelInterface, ResultInterface
                     let attributeName = attribute;
                 }
 
-                if hydrationMode == Resultset::HYDRATE_ARRAYS {
-                    let hydrateArray[attributeName] = value;
-                } else {
-                    let hydrateObject->{attributeName} = value;
-                }
+                let hydrateArray[attributeName] = value;
             } else {
-                if hydrationMode == Resultset::HYDRATE_ARRAYS {
-                    let hydrateArray[key] = value;
-                } else {
-                    let hydrateObject->{key} = value;
-                }
+                let hydrateArray[key] = value;
             }
         }
 
-        if hydrationMode == Resultset::HYDRATE_ARRAYS {
-            return hydrateArray;
+        if hydrationMode != Resultset::HYDRATE_ARRAYS {
+            return Arr::arrayToObject(hydrateArray);
         }
 
-        return hydrateObject;
+        return hydrateArray;
```

Comment:
This is actually slower than setting the object directly. With the current implementation we create a `stdClass` object and set its properties and return it in one loop while this implementation does two loops, one for the array and one to make it an object.

Reply:
Slightly slower but significantly less complicated.